### PR TITLE
feat(plasma-theme-builder): Add selectors from color palette for token value

### PR DESCRIPTION
**Pull request by neretin-trike**
Добавлена возможность выбора цвета токена из палитры (пакета `@salutejs/plasma-colors`). 

Добавлена возможность вставлять любое значение в поле для указания цвета.

Удалена генерация группы базовых цветов (baseColors) и сделан небольшой рефакторинг импортов.

:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.523.5059307314.swift)  
[color_sberHealth_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.523.5059307314.kt)  
[color_sberHealth_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.523.5059307314.ts)  
[color_sbermarket_business_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.523.5059307314.swift)  
[color_sbermarket_business_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.523.5059307314.kt)  
[color_sbermarket_business_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.523.5059307314.ts)  
[color_sbermarket_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.523.5059307314.swift)  
[color_sbermarket_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.523.5059307314.kt)  
[color_sbermarket_metro_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.523.5059307314.swift)  
[color_sbermarket_metro_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.523.5059307314.kt)  
[color_sbermarket_metro_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.523.5059307314.ts)  
[color_sbermarket_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.523.5059307314.ts)  
[color_sbermarket_selgros_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.523.5059307314.swift)  
[color_sbermarket_selgros_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.523.5059307314.kt)  
[color_sbermarket_selgros_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.523.5059307314.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.523.5059307314.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.523.5059307314.kt)  
[color_sbermarket_wlbusiness_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.523.5059307314.ts)  
[color_sberonline_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.523.5059307314.swift)  
[color_sberonline_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.523.5059307314.kt)  
[color_sberonline_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.523.5059307314.ts)  
[color_sberprime_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.523.5059307314.swift)  
[color_sberprime_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.523.5059307314.kt)  
[color_sberprime_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.523.5059307314.ts)  
[shadow_sbermarket_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.523.5059307314.ts)  
[typo_mage_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.523.5059307314.swift)  
[typo_mage_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.523.5059307314.kt)  
[typo_mage_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.523.5059307314.ts)  
[typo_plasma_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.523.5059307314.swift)  
[typo_plasma_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.523.5059307314.kt)  
[typo_plasma_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.523.5059307314.ts)  
[typo_ruler_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.523.5059307314.swift)  
[typo_ruler_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.523.5059307314.kt)  
[typo_ruler_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.523.5059307314.ts)  
[typo_sage_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.523.5059307314.swift)  
[typo_sage_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.523.5059307314.kt)  
[typo_sage_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.523.5059307314.ts)  
[typo_sbermarket_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.523.5059307314.swift)  
[typo_sbermarket_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.523.5059307314.kt)  
[typo_sbermarket_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.523.5059307314.ts)  
[typo_soulmate_ios-swift--canary.523.5059307314.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.523.5059307314.swift)  
[typo_soulmate_kotlin--canary.523.5059307314.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.523.5059307314.kt)  
[typo_soulmate_react-native--canary.523.5059307314.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.523.5059307314.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.216.0-canary.523.5059307314.0
  npm install @salutejs/plasma-temple@1.165.0-canary.523.5059307314.0
  npm install @salutejs/plasma-ui@1.197.0-canary.523.5059307314.0
  npm install @salutejs/plasma-web@1.216.0-canary.523.5059307314.0
  # or 
  yarn add @salutejs/plasma-b2c@1.216.0-canary.523.5059307314.0
  yarn add @salutejs/plasma-temple@1.165.0-canary.523.5059307314.0
  yarn add @salutejs/plasma-ui@1.197.0-canary.523.5059307314.0
  yarn add @salutejs/plasma-web@1.216.0-canary.523.5059307314.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-523/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-523/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-523/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-523/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-523/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-523/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-523/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx [hunk 1]
```diff
@@ -15,21 +17,31 @@ const StyledTokenForm = styled(Modal)`
...
     gap: 1rem;
+    flex-direction: column;
 `;
 
 const PreviewColor = styled.div<{ value: string }>`
     width: 3rem;
     height: 3rem;
     border-radius: 0.75rem;
-    background-color: ${({ value }) => value};
+    background: ${({ value }) => value};
```
Comment (Yakutoc) [inline website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:27]: [Уточнение]

@neretin-trike А почему решил так сделать, через background?

**Review by Yakutoc — APPROVED** [review]

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx (hunk 1 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:27]: просто теперь в качестве значений может прилететь `linear-gradient`, а он только на background сработает

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx [hunk 2]
```diff
@@ -54,45 +65,75 @@ export const TokenForm = ({
...
+
+    const onChangeName = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+
+        setName({ value });
+    }, []);
+
+    const onChangeSelectedColor = useCallback(
+        (color: GeneralColor) => {
+            const saturation500 = 7;
```
Comment (kayman233) [inline website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:85]: Уточнение: а что здесь происходит? Непонятно откуда 7

**Review by kayman233 — APPROVED** [review]

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx (hunk 2 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:85]: ну вот я как раз это число в отдельную переменную вытащил, думал станет чуть яснее :D Смысл в том, что массив с оттенками цветов от 50 до 1000 (с шагом 50), и 7 элемент в этом массиве - оттенок 500

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**